src/algorithms/sample-graph-data.ts

```diff
@@ -0,0 +1,215 @@
+import {INode, Node} from '../api/schema/node';
+import EdgeCost from './models/edgecost';
+import Graph from './models/graph';
+import Vertex from './models/vertex';
+
+const sampleGraph = new Graph();
+
+function getNode(id: number, x: number, y: number): INode {
+    return new Node({
```

Comment:
The Node you imported is a [Mongoose document class](https://mongoosejs.com/docs/models.html) and it is used for retrieving and updating data in the database. Even though it is probably working, it shouldn't be used like this at all. Make an instance of INode instead, we made this interface exactly for these purposes - to use the same Node definition in our code and Mongoose. But here you're importing a mongoose class and who know what's happening inside.

This might work, but I didn't try it:
```
return <INode>{
  ...
}
```